sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -271,6 +275,19 @@ static Builder builder() {
     abstract Builder setDefaultTableProvider(@Nullable String defaultTableProvider);
 
     abstract SqlTransform build();
+
+    @Override
+    public PTransform<PInput, PCollection<Row>> buildExternal(
+        External.Configuration configuration) {
+      return builder()
```

Comment:
@chamikaramj @chadrik It'd be nice to reduce this boilerplate to almost nothing, given we're already using the builder pattern here, or a minimal mapping of names (e.g. query -> queryString).

Reply:
I agree that a simple mapping would be nice. The current interface could still be useful for the more advanced external configurations. Does not seem to be of very high priority though.